src/docquery/document.py

```diff
@@ -167,38 +176,70 @@ def preview(self) -> "Image":
         return [self.b.convert("RGB")]
 
     @cached_property
-    def context(self) -> Tuple[(str, List[int])]:
-        words, boxes = apply_tesseract(self.b, lang=None, tesseract_config="")
+    def context(self) -> Dict[str, List[Tuple["Image.Image", List[Any]]]]:
+        words, boxes = self._normalize_boxes(*self.ocr_reader.apply_ocr(self.b), self.b.width, self.b.height)
+
         return {
             "image": [
                 (
                     self.b,
-                    [x for x in zip(words, boxes)],
+                    self._make_word_boxes(words, boxes),
                 )
             ]
         }
 
 
 @validate_arguments
-def load_document(fpath: str):
+def load_document(fpath: str, ocr_reader_name=None):
```

Comment:
I think it would be better for this and `ImageDocument` to both accept a `Union[str, OcrReader]` so that the user can flexibly pass in either a string or instantiated object in either location. I fat fingered this a few times while trying to use the methods. It would also allow us to have only one parameter name `ocr_reader` throughout.

Reply:
This was a pain with pydantic validator but I got it working! Much nicer to use now.